### ISSUE TITLE
cmd shell broken on 2008 sp2

### ISSUE THREAD

**Issue by mwrock** [WinRb/WinRM#232]
It is basically broken on any windows version prior to 2008R2/win 7. The powershell is fine (yay PSRP). Calling .bat/.cmd files essentially do nothing. The command immediately returns with no output but without invoking the script. The reason is codepage. We use 65001 (UTF-8) which:
1. Supports the broadest range of characters accross locales
2. Allows Windows Nano server to "work" (it returns a WSManFault otherwise)

However older OS's do not know how to UTF-8.

I'd like to discuss a few options to fix this:
1. Bring back the winrm v1 implementation of preceeding a `create_shell` with a WMI OS check and default to codepage `437` (ascii) for older OSs.
2. Go back to defaulting to `437` instead of 65001 (we used to default to 437 and changed to 65001 [a year ago](https://github.com/WinRb/WinRM/pull/130)). Then rescue on the WSManFault received on Nano and switch to 65001 and retry.
3. Allow users the ability to specify a codepage and if working with older OS's, expect users to know to use 437 (or another codepage appropriate to their locale).

I personally like 3 the most but here are some comments on each:
1. I don't like having to incur the WMI roundtrip for everyone just to appeal to 10 year old OSs.
2. A universal 437 default is not ideal (but we can customize) and this forces an extra round trip for Nano server on every shell create.
3. Its not obvious that commands do nothing because of codepage so this is not discoverable at all.

I'd like to get other's thoughts on this:

cc @sneal @smurawski @btm @MatthiasWinzeler

**Comment by sneal** [comment]
My preference is to default towards making newer OSs happy, not 10 year old ones. I would prefer dropping support for anything older than 2008R2, however downstream consumers like Chef probably can't easily do that.

Another option that came to mind just now, what about making the WinRM API ask the user for their target OS?

```
opts = { 
  endpoint: 'http://myhost:5985/wsman',
  user: 'administrator',
  password: '123456',
  os: :windows2012r2
}
conn = WinRM::Connection.new(opts)
```

Slightly more abstracted away from code page, and there's potential for other specific actions/config based on OS type.

**Comment by smurawski** [comment]
On the lines of graceful degradation suggested by @sneal (which I'm in favor of), we could alternatively interrogate the winrm endpoint get the wsman stack version.  if it is 2.0, you know you are on 2008r2 or earlier.  Also, if you make an authenticated request to the endpoint, you can get the OS version, which could be a better discriminator.

**Comment by mwrock** [comment]
fixed via #233
